provider/ec2/environ.go

```diff
@@ -583,7 +583,7 @@ func (e *environ) StartInstance(ctx context.ProviderCallContext, args environs.S
 	}
 
 	switch {
-	case subnetErr != nil && errors.IsNotFound(subnetErr):
+	case subnetErr != nil && (errors.IsNotFound(subnetErr) || isNotFoundError(err)):
```

Comment:
Shouldn't this be `isNotFoundError(subnetErr)`? Also, looking at the definition of `isNotFoundError` it already includes checking for `errors.IsNotFound`, so you should just be able to use that.